lib/vector-core/src/lib.rs

```diff
@@ -24,6 +24,7 @@
 #![allow(clippy::default_trait_access)]
 #![allow(clippy::enum_glob_use)]
 #![allow(clippy::unnested_or_patterns)]
+#![allow(clippy::match_wildcard_for_single_variants)]
```

Comment:
I actually like `match_wildcard_for_single_variants`. I think we should just be using `if let Some(_) = foo { } else { }` instead.

Reply:
Hmm, so I think the `if let` pattern is applicable when we don't capture the result in the wildcard match but that's not totally consistent. We would still need places where either we make a site exception for wildcard matching or match all variants. I'd be glad to see that discussed in a follow-up PR, that said.

Reply:
🤔 I'm not sure I completely follow. `match_wildcard_for_single_variants` is only flagged when the wildcard, `_`, matches a single variant. I think we should be able to use `if let Some()` for all of those? I think when we are `match`ing on two variants where one is a `_` indicates we are only ever going to care about the single variant we are actually matching on.

Reply:
Aha, nvm, I think I see what you are saying. When we actually need the result of the `match`. I think you could still do that with:

```rust
let foo = if let Some(v) = bar {
  v
} else {
  # whatever
}
```

But I take your point. I think the `match` is clearer in that case. 👍

Reply:
I'm going to mark this as resolved but I'd for sure be happy to talk about this in a follow-up PR.